pkg/scheduler/scheduler_test.go

```diff
@@ -2299,14 +2301,29 @@ func TestSchedulerSchedulePod(t *testing.T) {
 				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
 				st.RegisterPreFilterPlugin(
 					"FakePreFilter",
-					st.NewFakePreFilterPlugin(framework.NewStatus(framework.Error, "injected error status")),
+					st.NewFakePreFilterPlugin(nil, framework.NewStatus(framework.Error, "injected error status")),
 				),
 				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
 			},
-			nodes:         []string{"1", "2"},
-			pod:           &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
-			expectedHosts: nil,
-			wErr:          fmt.Errorf(`running PreFilter plugin "FakePreFilter": %w`, errors.New("injected error status")),
+			nodes:     []string{"1", "2"},
+			pod:       &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+			wantNodes: nil,
+			wErr:      fmt.Errorf(`running PreFilter plugin "FakePreFilter": %w`, errors.New("injected error status")),
+		},
+		{
+			name: "test prefilter plugin returning node",
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterPreFilterPlugin(
+					"FakePreFilter",
+					st.NewFakePreFilterPlugin(&framework.PreFilterResult{NodeNames: sets.NewString("node2")}, nil),
+				),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			},
+			nodes:              []string{"node1", "node2", "node3"},
+			pod:                &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-prefilter", UID: types.UID("test-prefilter")}},
+			wantNodes:          sets.NewString("node2"),
+			wantEvaluatedNodes: pointer.Int32Ptr(1),
```

Comment:
Before this PR, preFilter didn't evaluate nodes. So the EvaluatedNodes here is equal to the evaluated nodes in the filter phase. But after this PR merged, we need to change the comments below to express EvaluatedNodes comes just from the filter phase. WDYT?
```go
// ScheduleResult represents the result of one pod scheduled. It will contain
// the final selected Node, along with the selected intermediate information.
type ScheduleResult struct {
	// Name of the scheduler suggest host
	SuggestedHost string
	// Number of nodes scheduler evaluated on one pod scheduled < on filter phase >
	EvaluatedNodes int
	// Number of feasible nodes on one pod scheduled
	FeasibleNodes int
}
```

Reply:
Sorry, I didn't express myself clearly. I means the comment is better to changed like this
`// Number of nodes scheduler evaluated on one pod scheduled` ==>
`// Number of nodes scheduler evaluated on one pod scheduled on filter phase `

Reply:
updated